doc/modules/grid_search.rst

```diff
@@ -150,6 +154,217 @@ increasing ``n_iter`` will always lead to a finer search.
       Random search for hyper-parameter optimization,
       The Journal of Machine Learning Research (2012)
 
+.. _successive_halving_user_guide:
+
+Searching optimal parameters with successive halving
+====================================================
+
+Scikit-learn also provides the :class:`GridHalvingSearchCV` and
+:class:`RandomHalvingSearchCV` estimators that can be used to
+search a parameter space using successive halving [1]_ [2]_. Successive
+halving is an iterative selection process where all candidates are evaluated
+with a small amount of resources at the first iteration. Only some of
+these candidates are selected for the next iteration, which will be
+allocated more resources. What defines a resource is typically the number of
+samples to train on, or the number of trees for a gradient boosting /
+decision forest estimator.
+
+As illustrated in the figure below, only a small subset of candidates 'survive'
+until the last iteration. These are the candidates that have consistently been
+part of the best candidates across all iterations.
+
+.. figure:: ../auto_examples/svm/images/sphx_glr_plot_successive_halving_iterations_001.png
+   :target: ../auto_examples/model_selection/plot_successive_halving_iterations.html
+   :align: center
+
+The amount of resources ``r_i`` allocated for each candidate at iteration
+``i`` is controlled by the parameters ``ratio`` and ``r_min`` as follows::
+
+    r_i = ratio**i * r_min
+
+``r_min`` is the amount of resources used at the first iteration and
+``ratio`` defines the proportions of candidates that will be selected for
+the next iteration::
+
+    n_candidates_to_keep = n_candidates_at_i // ratio
+
+So in the first iteration, we use ``r_min`` resources ``n_candidates``
+times. In the second iteration, we use ``r_min * ratio`` resources
+``n_candidates // ratio`` times. The third again multiplies the resources
+per candidate and divides the number of candidates. This process stops when
+the maximum budget per candidate is reached, or when less than ``ratio``
+candidates are left.
```

Comment:
Either here, or maybe even earlier, I would talk about how the number of candidates relates to the number of iterations and give an example, maybe using the table below.
In our implementation, the number of candidates is actually inferred using r_min and the number of iterations.

So I would make the example something like
"Assume r_min is 10 and we have a dataset of 10000 samples. That allows us to run X iterations of successive halving. That means if we start with ... candidates, we will end up with n_ratio candidates exactly when we made use of the whole dataset.
So in the case of using the dataset as a budget, it makes sense to automatically determine the number of candidates. These are usually sampled randomly from a distribution of candidate parameters.
Using more candidates will result in more models being trained on the full dataset, meaning the algorithm will perform closer to standard RandomizedSearchCV, while using less candidates will result in not all of the data being used.
"


something like that?
And then go over the case of using ``n_iterations`` and ``n_estimators`` again?

Maybe it would be nice to also discuss the idea of the brackets as they are used in hyper-band. I.e. if we need a lot of resources to figure out which parameter setting is best, we need to run a lot of configurations to the end, so something like RandomizedSearchCV is better (or equivalently having a high r_min given a particular number of candidates).

If we only need very little resources to distinguish good and bad parameters, then our pretty aggressive schedule from above, that only runs ratio many configurations on the full dataset might be enough and is much faster.